taxi/views.py

```diff
@@ -57,12 +67,38 @@ class ManufacturerDeleteView(LoginRequiredMixin, generic.DeleteView):
 class CarListView(LoginRequiredMixin, generic.ListView):
     model = Car
     paginate_by = 5
-    queryset = Car.objects.select_related("manufacturer")
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super(CarListView, self).get_context_data(**kwargs)
+        model = self.request.GET.get("model", "")
+        context["search_form"] = CarSearchForm(initial={"model": model})
+        return context
+
+    def get_queryset(self):
+        queryset = Car.objects.select_related("manufacturer")
+        form = CarSearchForm(self.request.GET)
+        if form.is_valid():
+            return queryset.filter(model__icontains=form.cleaned_data["model"])
+        return queryset
 
 
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super(CarDetailView, self).get_context_data(**kwargs)
+        username = self.request.GET.get("username")
+        context["search_form"] = DriverSearchForm(
+            initial={"username": username}
+        )
+        return context
+
+    def get_queryset(self):
+        username = self.request.GET.get("username")
+        if username:
+            return Driver.objects.filter(username__icontains=username)
+        return Driver.objects.all()
```

Comment:
The `get_queryset` method in `CarDetailView` is filtering `Driver` objects based on the `username` query parameter. This seems incorrect as the `CarDetailView` should be dealing with `Car` objects. Consider revising this logic to ensure it aligns with the intended functionality of the view.